goth/runner/probe/__init__.py

```diff
@@ -332,8 +336,8 @@ async def run_command_on_host(
 
         finally:
             await cmd_monitor.stop()
-            logger.debug("Waiting for the command to finish")
-            await asyncio.gather(cmd_task, return_exceptions=True)
+            for assertion in cmd_monitor.failed:
+                raise TemporalAssertionError(assertion.name)
```

Comment:
Let's not forget to report assertion failures that occurred at the "end of events".